tools/clang/lib/SPIRV/TypeTranslator.cpp

```diff
@@ -148,6 +148,14 @@ uint32_t TypeTranslator::translateType(QualType type, LayoutRule rule,
           return theBuilder.getFloat32Type();
         case BuiltinType::Double:
           return theBuilder.getFloat64Type();
+        case BuiltinType::LitFloat: {
+          const auto &semantics = astContext.getFloatTypeSemantics(type);
+          const auto bitwidth = llvm::APFloat::getSizeInBits(semantics);
+          if (bitwidth <= 32)
```

Comment:
We should probably check this is exactly 32 bit or 64 bit and error out for other cases. Numbers are sensitive.

Reply:
There are other types (Int, Short, and Min12Int) which should also translate to 32-bit integer.
I doubt that the getSizeInBits returns anything other than known integer type widths. And if the bitwidth is anything larger than 32, (say 46), then we still need to use a 64-bit integer to represent it.

Reply:
But this is for floats? But reasonable since it is actually that how we use literal floats determines the number of bits it should take.